File: update_lpr.js
const fs = require('fs');
const path = require('path');

const csvPath = 'C:\\Users\\gparra\\MapaTraficoFinal\\public\\Camaras.CSV1.csv';
const geojsonPath = 'C:\\Users\\gparra\\TraficoMapGeneral\\public\\data\\cameras.geojson';

// Read CSV and extract LPR cameras
const csvData = fs.readFileSync(csvPath, 'utf-8');
const lines = csvData.split('\n').slice(1); // Skip header

const lprCameras = new Set();

lines.forEach(line => {
  if (!line.trim()) return;
  
  // CSV parsing (simple approach for quoted fields)
  const parts = [];
  let current = '';
  let inQuotes = false;
  
  for (let i = 0; i < line.length; i++) {
    const char = line[i];
    if (char === '"') inQuotes = !inQuotes;
    else if (char === ',' && !inQuotes) {
      parts.push(current.trim());
      current = '';
    } else {
      current += char;
    }
  }
  parts.push(current.trim());
  
  // Columns: Longitud(1), Latitud(2), Direccion(3), N CAMARA(4), Domos(5), Fijas(6), LPR(7)...
  const nCamara = parts[4] ? parts[4].replace(/"/g, '').trim() : '';
  const lpr = parts[7] ? parts[7].replace(/"/g, '').trim() : '';
  
  if (nCamara && lpr) {
    // Has both camera number and LPR value
    try {
      lprCameras.add(parseInt(nCamara));
    } catch (e) {
      //ignore
    }
  }
});

console.log(`Found ${lprCameras.size} LPR cameras`);
console.log(`List: ${Array.from(lprCameras).sort((a,b) => a-b).join(', ')}`);

// Update GeoJSON
const geojson = JSON.parse(fs.readFileSync(geojsonPath, 'utf-8'));

const before = geojson.features.filter(f => f.properties.lpr === 1).length;

geojson.features.forEach(feature => {
  const camNum = parseInt(feature.properties.camera_number);
  if (lprCameras.has(camNum)) {
    feature.properties.lpr = 1;
  }
});

const after = geojson.features.filter(f => f.properties.lpr === 1).length;

fs.writeFileSync(geojsonPath, JSON.stringify(geojson, null, 2));

console.log(`Before: ${before} LPRs`);
console.log(`After: ${after} LPRs`);
console.log(`✅ Updated!`);

File: update_lpr.py
#!/usr/bin/env python
"""Update cameras.geojson with LPR data from MapaTraficoFinal CSV"""
import csv
import json

def main():
    # Find all unique LPR camera numbers from source CSV
    lpr_cameras = set()
    csv_path = r'C:\Users\gparra\MapaTraficoFinal\public\Camaras.CSV1.csv'
    
    print("📖 Reading MapaTraficoFinal CSV...")
    try:
        with open(csv_path, 'r', encoding='utf-8') as f:
            reader = csv.DictReader(f)
            for row in reader:
                lpr_value = row.get('LPR', '').strip()
                if lpr_value:  # Not empty
                    camera_num = row.get('N CAMARA', '').strip()
                    if camera_num:
                        try:
                            lpr_cameras.add(int(camera_num))
                        except ValueError:
                            pass
    except Exception as e:
        print(f"❌ Error reading CSV: {e}")
        return False
    
    print(f"✅ Found {len(lpr_cameras)} unique LPR cameras")
    print(f"   List: {sorted(list(lpr_cameras))}")
    
    # Update cameras.geojson
    geojson_path = r'C:\Users\gparra\TraficoMapGeneral\public\data\cameras.geojson'
    
    print("\n🔄 Updating cameras.geojson...")
    try:
        with open(geojson_path, 'r', encoding='utf-8') as f:
            geojson_data = json.load(f)
        
        before = sum(1 for f in geojson_data['features'] if f.get('properties', {}).get('lpr') == 1)
        
        updated = 0
        for feature in geojson_data['features']:
            props = feature.get('properties', {})
            try:
                camera_num = int(props.get('camera_number', 0))
                if camera_num in lpr_cameras and props.get('lpr') != 1:
                    props['lpr'] = 1
                    updated += 1
            except (ValueError, TypeError):
                pass
        
        after = sum(1 for f in geojson_data['features'] if f.get('properties', {}).get('lpr') == 1)
        
        with open(geojson_path, 'w', encoding='utf-8') as f:
            json.dump(geojson_data, f, ensure_ascii=False, indent=2)
        
        print(f"   Before: {before} LPR cameras")
        print(f"   Updated: {updated} cameras")
        print(f"   After: {after} LPR cameras")
        print(f"✅ cameras.geojson updated successfully!")
        return True
        
    except Exception as e:
        print(f"❌ Error updating GeoJSON: {e}")
        return False

if __name__ == '__main__':
    success = main()
    exit(0 if success else 1)
